pkg/edgeview/src/system.go

```diff
@@ -938,6 +967,213 @@ func dispAFile(f os.FileInfo) {
 	fmt.Printf("%s, %v, %d, %s\n", f.Mode().String(), f.ModTime(), f.Size(), f.Name())
 }
 
+func getTarFile(opt string) {
+	execCmd := strings.SplitN(opt, "tar/", 2)
+	if len(execCmd) != 2 {
+		fmt.Printf("tar needs a / and path input\n")
+		return
+	}
+	source := execCmd[1]
+	printColor(" - tar cmd: "+source, colorCYAN)
+	closePipe(true)
+
+	fi, err := os.Stat(source)
+	if err != nil {
+		fmt.Printf("find directory error: %v\n", err)
+		return
+	}
+
+	if !fi.IsDir() {
+		fmt.Printf("the path %s is not a directory\n", source)
+		return
+	}
+
+	ok := checkBlockedDirs(source)
+	if !ok {
+		fmt.Printf("the directory %s can not be tarred\n", source)
+		return
+	}
+
+	// check the directory size not to exceed the max
+	sizeBytes := du(source, fi)
+	if sizeBytes > tarMaxBytes {
+		fmt.Printf("directory size %d, exceeds maximum %d\n", sizeBytes, tarMaxBytes)
+		return
+	}
+	baseDir := filepath.Base(source)
+	targetDir := "/tmp/"
+	dirs := strings.Split(source, "/")
+	tarfileName := targetDir
+	for _, d := range dirs {
+		if d == "" {
+			continue
+		}
+		tarfileName = tarfileName + d + "."
+	}
+	tarfileName = tarfileName + strconv.FormatInt(time.Now().Unix(), 10) + ".tar"
+	tarFile, err := os.Create(tarfileName)
+	if err != nil {
+		log.Errorf("can not create tar file")
+		return
+	}
+	defer tarFile.Close()
+
+	tarball := tar.NewWriter(tarFile)
+	defer tarball.Close()
+
+	err = filepath.Walk(source,
+		func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+			// skip file with suffix of '.key.pem' or '.key'
+			ok := checkBlockedFileSuffix(info.Name())
```

Comment:
Need to check you don't descend in /persist/vault etc here as well I think.

Reply:
the /persist/vault is excluded before getting to here already.

Reply:
Only if the argument is /persist/vault.

Reply:
ok.